autobump-core/src/main/java/com/github/autobump/core/model/usecases/AutobumpUseCase.java

```diff
@@ -34,36 +34,68 @@
+                           DependencyResolver dependencyResolver,
+                           VersionRepository versionRepository,
+                           DependencyBumper dependencyBumper,
+                           UrlHelper urlHelper,
+                           URI uri) {
+        this.gitProvider = gitProvider;
+        this.gitClient = gitClient;
+        this.dependencyResolver = dependencyResolver;
+        this.versionRepository = versionRepository;
+        this.dependencyBumper = dependencyBumper;
+        this.urlHelper = urlHelper;
+        this.uri = uri;
+    }
 
     public AutobumpResult doAutoBump() {
-        int amountOfBumps = 0;
         Workspace workspace = gitClient.clone(getUri());
-        for (Dependency dependency : dependencyResolver.resolve(workspace)) {
-            Version latestVersion = getLatestVersion(dependency);
-            if (latestVersion != null &&
-                    dependency.getVersion().compareTo(latestVersion) > 0) {
-                Bump bump = BumpUseCase.builder()
-                        .dependency(dependency)
-                        .dependencyBumper(dependencyBumper)
-                        .latestVersion(latestVersion)
-                        .workspace(workspace)
-                        .build()
-                        .doBump();
+        var dependencymap = ResolveDependenciesUseCase.builder()
+                .dependencyResolver(dependencyResolver)
+                .workspace(workspace)
+                .build()
+                .deResolve();
+        makeBumpsAndPullRequests(workspace, dependencymap);
+        return new AutobumpResult(amountofbumps);
+    }
+
+    private void makeBumpsAndPullRequests(Workspace workspace, Map<String,
+            Set<Dependency>> dependencymap) {
+        for (var entry : dependencymap.entrySet()) {
+            boolean bumped = false;
+            for (Dependency dependency : dependencymap.get(entry.getKey())) {
+                Version latestVersion = getLatestVersion(dependency);
+                if (latestVersion != null &&
+                        dependency.getVersion().compareTo(latestVersion) > 0) {
+                    BumpUseCase.builder()
+                            .dependency(dependency)
+                            .dependencyBumper(dependencyBumper)
+                            .latestVersion(latestVersion)
+                            .workspace(workspace)
+                            .build().doBump();
+                    bumped = isBumped();
+                }
+            }
+            if (bumped) {
                 PullRequestUseCase.builder()
-                        .gitProvider(gitProvider)
-                        .gitClient(gitClient)
-                        .urlHelper(urlHelper)
-                        .workspace(workspace)
-                        .bump(bump)
-                        .uri(uri)
-                        .build()
+                        .gitProvider(gitProvider).gitClient(gitClient)
```

Comment:
I suppose this method become too long for code quality checks and the builder was adapted to included multiple method calls on one line. It might have been better to extract some methods I believe.